pkg/antctl/raw/multicluster/join.go

```diff
@@ -0,0 +1,277 @@
+	}
+	if o.LeaderClusterOpt.Secret == "" && o.Secret == nil {
+		return fmt.Errorf("specify at least one of the leader cluster Secret and the Secret in the config file")
+	}
+	if o.LeaderClusterOpt.Namespace == "" {
+		return fmt.Errorf("the the leader cluster Namespace is required")
+	}
+	if o.ClusterSetID == "" {
+		return fmt.Errorf("the ClusterSet ID is required")
+	}
+	if o.MemberClusterOpt.ClusterID == "" {
+		return fmt.Errorf("the member clusterID is required")
+	}
+	if o.MemberClusterOpt.Namespace == "" {
+		return fmt.Errorf("the member cluster Namespace cannot is required")
+	}
+
+	return nil
+}
+
+func yamlUnmarshall(raw []byte, v interface{}) error {
+	d := yaml.NewDecoder(bytes.NewReader(raw))
+	d.KnownFields(false)
+	return d.Decode(v)
+}
+
+func unmarshallSecret(raw []byte) (*v1.Secret, error) {
+	decoder := yamlutil.NewYAMLOrJSONDecoder(bytes.NewReader(raw), 100)
+	secret := &v1.Secret{}
+	// The config file is a YAML file, so we need to decode it to a Secret.
+	// And the first object in the file is the ClusterSetConfig, so we need to skip it.
+	u := unstructured.Unstructured{}
+	if err := decoder.Decode(&u); err != nil {
+		return nil, err
+	}
+	if err := decoder.Decode(secret); err != nil {
+		return nil, err
+	}
+
+	return secret, nil
+}
+
+var joinExamples = strings.Trim(`
+# Join the ClusterSet from a member cluster with command line options with secret.
+  $ antctl mc join --leader-namespace=antrea-multicluster \
+				   --leader-clusterid=cluster-north \
+				   --secret=member-access-token \
+				   --leader-server=https://172.18.0.3:6443 \
+				   --clusterset=clusterset \
+				   --namespace=kube-system \
+				   --clusterid=cluster-east
+
+# Join the ClusterSet from a member cluster with a config file. 
+  $ antctl mc join --config-file ./config.yaml
+
+# Config file example:
+apiVersion: multicluster.antrea.io/v1alpha1
+kind: ClusterSetConfig
+memberCluster:
+  namespace: kube-system
+  clusterID: cluster-east
+leaderCluster:
+  namespace: antrea-multicluster
+  secret: member-access-token
+  apiServer: https://172.18.0.3:6443
+  clusterID: cluster-north
+clusterSetID: clusterset
+`, "\n")
+
+func NewJoinCommand() *cobra.Command {
+	command := &cobra.Command{
+		Use:     "join",
+		Short:   "Join the ClusterSet from a member cluster",
+		Args:    cobra.MaximumNArgs(0),
+		Example: joinExamples,
+		RunE:    joinRunE,
+	}
+
+	o := JoinOptions{}
+	joinOpts = &o
+	command.Flags().StringVarP(&joinOpts.LeaderClusterOpt.Namespace, "leader-namespace", "", "", "Namespace of the leader cluster")
+	command.Flags().StringVarP(&joinOpts.LeaderClusterOpt.ClusterID, "leader-clusterid", "", "", "Cluster ID of the leader cluster")
+	command.Flags().StringVarP(&joinOpts.LeaderClusterOpt.Secret, "token-secret-name", "", "", "Name of the Secret resource that contains the leader cluster token")
+	command.Flags().StringVarP(&joinOpts.LeaderClusterOpt.APIServer, "leader-server", "", "", "API Server endpoint of the leader cluster")
```

Comment:
leader-apiserver to be consistent with the config file field.

Reply:
done.